pkl-gradle/pkl-gradle.gradle.kts

```diff
@@ -71,3 +69,29 @@ signing {
     }
   }
 }
+
+tasks.compileTestKotlin {
+  // Work around a conflict between Pkl's and Gradle's
+  // Kotlin dependencies on the test compile class path.
+  //
+  // My preferred solution would be to clean up the test 
+  // compile class path to no longer contain a Gradle distribution. 
+  // However, my Gradle knowledge proved insufficient to accomplish this.
+  //
+  // Another potential workaround is to port plugin tests to Java.
+  // (If the plugin was written in Kotlin, its compilation would 
+  // currently fail with the same error.)
```

Comment:
> I don't think we can eschew Gradle's distribution from the test compile classpath, because our tests necessarily include Gradle's SDKs.

We can, if we can figure out how to implement this in Gradle.
These tests just run builds with `GradleRunner`.
`GradleRunner` is similar to Pkl's `Executor` in that it loads Gradle in a separate class loader.
Unless `withDebug()` is used, it even runs Gradle in a separate JVM.

I believe Gradle JARs and their dependencies end up on the test compile class path because
the Gradle Gradle plugin declares them as `implementation` (or even `api`) dependencies`,
which the `testCompileClassPath` configuration inherits from.
In other words, this looks like a bug/sloppiness of the Gradle Gradle plugin that
is unlikely to surface unless plugin tests are written in Kotlin.

> But, we can get around this issue if/when we upgrade to Kotlin 1.9 too.

This cannot be relied upon. The next time Gradle or Pkl updates their Kotlin version, the version skew will be back.

Reply:
Okay, I played around with this a little bit more, and this works without needing to add a `-Xskip-metadata-version-check` compiler flag.

```
sourceSets {
  test {
    // Exclude Gradle's kotlin stdlib from test compile classpath (conflicts with project's kotlin)
    compileClasspath = compileClasspath.filter { !it.path.matches(Regex(".*gradle-\\d+\\.\\d+/lib/kotlin-stdlib.*")) }
  }
}
```

I couldn't figure out how to exclude this by matching on group or artifact id. Seems like Gradle injects these deps out-of-band of the normal dependency structure. This seems related: https://github.com/gradle/gradle/issues/19973

Might be interesting to try out https://github.com/gradle-plugins/toolbox at some point

Reply:
Well done. I replaced my workaround with an improved version of your solution that removes all Gradle distribution JARs. Ready to merge from my side.